extensions/panache/hibernate-orm-panache-rest/runtime/src/main/java/io/quarkus/hibernate/orm/panache/rest/runtime/utils/StringUtil.java

```diff
@@ -0,0 +1,91 @@
+package io.quarkus.hibernate.orm.panache.rest.runtime.utils;
+
+import static io.quarkus.runtime.util.StringUtil.camelHumpsIterator;
+import static io.quarkus.runtime.util.StringUtil.lowerCase;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.StringJoiner;
+
+public final class StringUtil {
+
+    private static final List<String> REQUIRE_ES = Arrays.asList("s", "sh", "ch", "x", "z", "o");
+
+    private static final Map<String, String> IRREGULARS = new HashMap<>();
+
+    static {
+        IRREGULARS.put("child", "children");
+        IRREGULARS.put("goose", "geese");
+        IRREGULARS.put("man", "men");
+        IRREGULARS.put("woman", "women");
+        IRREGULARS.put("tooth", "teeth");
+        IRREGULARS.put("foot", "feet");
+        IRREGULARS.put("mouse", "mice");
+        IRREGULARS.put("person", "people");
+        IRREGULARS.put("photo", "photos");
+        IRREGULARS.put("piano", "pianos");
+        IRREGULARS.put("halo", "halos");
+        IRREGULARS.put("roof", "roofs");
+        IRREGULARS.put("belief", "beliefs");
+        IRREGULARS.put("chef", "chefs");
+        IRREGULARS.put("chief", "chiefs");
+    }
+
+    public static String camelToHyphenated(String camelString) {
+        if (camelString == null) {
+            return null;
+        }
+
+        StringJoiner joiner = new StringJoiner("-");
+        Iterator<String> strings = lowerCase(camelHumpsIterator(camelString));
+
+        while (strings.hasNext()) {
+            joiner.add(strings.next());
+        }
+
+        return joiner.toString();
+    }
+
+    public static String toPlural(String singular) {
```

Comment:
I'm not very found of having a method to compute the plural rules, maybe we need to ask the user to give us the root path of the controller istead of generating it automatically by computing the plural of the name of the entity. After all, having the plural of the name of the entity as a root path is a convention that some may not want to use.

And also, we only covers plurals in english and a lot of people are not using english inside their URL. So we may end up with a lot of feature request to support multiple languages ...

Reply:
I think it's a good default to have which would cover the majority of the cases. I'll add an option to override it though to `@PanacheRestResource` annotation for the next iteration. WDYT?

Reply:
I like supporting creation of urls by default in english. Later we can add support for multi-language pluralisation rules or having the user specify them entirely.

Reply:
Please, at least let the user specify it ...